### PR TITLE
Support formatting DateTime fields via AlloySerializationConfig

### PR DESCRIPTION
**Pull request by mrudula-gs**
#### What type of PR is this?
Improvement

#### What does this PR do / why is it needed ?
Users can configure formatting the DateTime properties on SerializationConfig. This is particularly important when the returned JSON is being sent to 3rd party systems where certain formatting is required.

#### Does this PR introduce a user-facing change?
No

### CODE REVIEW

File: legend-engine-pure-code-compiled-core/src/main/resources/core/pure/corefunctions/dateExtension.pure [hunk 1]
```diff
@@ -589,4 +589,10 @@ function meta::pure::functions::date::toEpochValue(d:Date[1]):Integer[1]
 function meta::pure::functions::date::toEpochValue(d:Date[1], unit:DurationUnit[1]):Integer[1]
 {
    dateDiff(%1970-01-01T00:00:00.000+0000, $d, $unit);
-}
\ No newline at end of file
+}
+
+function meta::pure::functions::date::validateDateTimeFormat(dateTimeFormat:String[1]):Boolean[1]
```
Comment (hardikmaheshwari) [inline legend-engine-pure-code-compiled-core/src/main/resources/core/pure/corefunctions/dateExtension.pure:594]: Are we restricting to only few possible timezone's?

File: legend-engine-pure-code-compiled-core/src/main/resources/core/pure/graphFetch/graphFetch.pure [hunk 2]
```diff
@@ -78,16 +78,27 @@ function meta::pure::graphFetch::execution::checkCompatibility(fe: FunctionExpre
...
+    {doc.doc = 'Give a valid string of dateTime format to serialize the dateTime properties respectively'}
+    dateTimeFormat: String[0..1];
+
     removePropertiesWithNullValues: Boolean[0..1];
     removePropertiesWithEmptySets: Boolean[0..1];
+
+    {doc.doc = 'Set to true to include fully qualified type info of the serialized class'}
     fullyQualifiedTypePath: Boolean[0..1];
+
+    {doc.doc = 'Set to True to include Alloy Relational Store Object Reference'}
```
Comment (hardikmaheshwari) [inline legend-engine-pure-code-compiled-core/src/main/resources/core/pure/graphFetch/graphFetch.pure:101]: lets change this to
'Set to True to include Alloy Store Object Reference'

**Review by hardikmaheshwari — CHANGES_REQUESTED** [review]

File: legend-engine-pure-code-compiled-core/src/main/resources/core/pure/graphFetch/graphFetch.pure (hunk 2 above)
Comment (mrudula-gs) [reply legend-engine-pure-code-compiled-core/src/main/resources/core/pure/graphFetch/graphFetch.pure]: Done

File: legend-engine-pure-code-compiled-core/src/main/resources/core/pure/corefunctions/dateExtension.pure (hunk 1 above)
Comment (mrudula-gs) [reply legend-engine-pure-code-compiled-core/src/main/resources/core/pure/corefunctions/dateExtension.pure:594]: Yes, we are restricting to the ones that are formattable by PureDate using regex

Comment (hardikmaheshwari) [reply legend-engine-pure-code-compiled-core/src/main/resources/core/pure/corefunctions/dateExtension.pure:594]: So we can only understand couple of timezones?

Comment (mrudula-gs) [reply legend-engine-pure-code-compiled-core/src/main/resources/core/pure/corefunctions/dateExtension.pure:594]: Updated the regex to all possible timezones

**Review by hardikmaheshwari — APPROVED** [review]

**Pull request merged**